### PR TITLE
[SPARK-27964][SQL] Move v2 catalog update methods to CatalogV2Util

### PR DESCRIPTION
**Pull request by rdblue**
## What changes were proposed in this pull request?

Move methods that implement v2 catalog operations to CatalogV2Util so they can be used in #24768.

## How was this patch tested?

Behavior is validated by existing tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for making this first, @rdblue !

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/utils/CatalogV2Util.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+object CatalogV2Util {
+  /**
+   * Apply properties changes to a map and return the result.
+   */
+  def applyPropertiesChanges(
+      properties: Map[String, String],
+      changes: Seq[TableChange]): Map[String, String] = {
+    applyPropertiesChanges(properties.asJava, changes).asScala.toMap
+  }
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/utils/CatalogV2Util.scala]: This is a newly added function during migration.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/utils/CatalogV2Util.scala [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        case _ =>
+          // ignore non-schema changes
+          schema
+      }
+    }
+  }
+
+  private def replace(
+      struct: StructType,
+      fieldNames: Seq[String],
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/utils/CatalogV2Util.scala]: Thank you for the update. (This field name is modified based on the recent comment.)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

This PR created `CatalogV2Util` by moving old `TestTableCatalog` object from test suites to `main` package `org.apache.spark.sql.catalog.v2.utils`. We will use this functions to implement `V2SessionCatalog`. All functions are moved correctly. And, one Scala-version function is added and one parameter is renamed according to the previous comments.

cc @gatorsmile

**Comment by gatorsmile** [comment]
I might need more time to understand the proposed V2SessionCatalog. The code movement in this PR is fine to me.

**Comment by dongjoon-hyun** [comment]
Thank you, @gatorsmile .

**Pull request closed without merging**